plugin/etcd/cname_test.go

```diff
@@ -58,6 +58,7 @@ var servicesCname = []*msg.Service{
 	{Host: "cname6.region2.skydns.test", Key: "cname5.region2.skydns.test."},
 	{Host: "endpoint.region2.skydns.test", Key: "cname6.region2.skydns.test."},
 	{Host: "mainendpoint.region2.skydns.test", Key: "region2.skydns.test."},
+	{Host: "", Key: "region3.skydns.test.", Text: "SOME-RECORD-TEXT"},
```

Comment:
wouldn't this be also true for other types well? I.e A query when there's only a txt record?

pr looks good btw, but this might be a bigger issue than just for CNAME queries.

Reply:
... I just tried to add a TXT service to the `lookup_test`, and it does indeed seem like the same thing happens there. The record gets included in the Answer section as a CNAME record pointing to `.`

Reply:
I actually have another fix now that should also cover this case for A, AAAA, and SRV lookups. It's a fairly crude check if the host is empty or not, but it seems to be working. Would you prefer that I push that here, or should I open a new PR for that change?